lib/src/json_serializable/type_helper.dart

```diff
@@ -0,0 +1,83 @@
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/dart/element/type.dart';
+
+abstract class TypeHelper {
+  const TypeHelper();
+
+  bool matchesTypeForSerialize(DartType type);
+
+  bool canDeserialize(DartType type);
+
+  String toSerializedValue(DartType searchType, String name);
+
+  String toDeserializedValue(DartType searchType, String expression);
+}
+
+class JsonHelper extends TypeHelper {
+  const JsonHelper();
+
+  @override
+  bool matchesTypeForSerialize(DartType type) {
+    //TODO(kevmoo): This should be checking for toJson method, but
+    //              toJson might be gone during generation, so check for
+    //              the annotation, too!
+    if (type is! InterfaceType) return false;
```

Comment:
Since this is identical to `canDeserialize` right now I would just call it - that also makes it more clear what this workaround is doing.